presto-main/src/main/java/com/facebook/presto/type/DecimalSaturatedFloorCasts.java

```diff
@@ -82,9 +88,13 @@ public static Slice longDecimalToLongDecimal(Slice value, int sourcePrecision, i
 
     private static BigInteger bigintToBigintFloorSaturatedCast(BigInteger value, int sourceScale, int resultPrecision, int resultScale)
     {
-        BigDecimal bigDecimal = new BigDecimal(value, sourceScale);
-        bigDecimal = bigDecimal.setScale(resultScale, FLOOR);
-        BigInteger unscaledValue = bigDecimal.unscaledValue();
+        return bigDecimalToBigintFloorSaturatedCast(new BigDecimal(value, sourceScale), resultPrecision, resultScale);
+    }
+
+    private static BigInteger bigDecimalToBigintFloorSaturatedCast(BigDecimal bigDecimal, int resultPrecision, int resultScale)
+    {
+        BigDecimal rescaledValue = bigDecimal.setScale(resultScale, FLOOR);
+        BigInteger unscaledValue = rescaledValue.unscaledValue();
         BigInteger maxUnscaledValue = bigIntegerTenToNth(resultPrecision).subtract(ONE);
```

Comment:
I think it is better express what i want to do here. I would like to keep it as is.